src/main/java/io/temporal/internal/sync/ActivityExecutionContextImpl.java

```diff
@@ -130,7 +130,7 @@
         return result;
       }
       Optional<Payloads> details = info.getHeartbeatDetails();
-      return Optional.ofNullable(dataConverter.fromPayloads(details, detailsClass, detailsType));
+      return Optional.ofNullable(dataConverter.fromPayloads(0, details, detailsClass, detailsType));
```

Comment:
we didn't want this function to have an overload without the index that internally assumes the first element?

Reply:
This class is used only internally by the SDK, so I think having an explicit index makes sense to indicate that it can hold multiple values.